api/src/main/java/org/openmrs/module/openconceptlab/client/OclClient.java

```diff
@@ -125,13 +132,73 @@ public GetMethod executeExportRequest(String url, String latestVersion) throws I
 		return exportUrlGet;
 	}
 
-	public OclResponse fetchLastReleaseVersion(String url, String token, String lastReleaseVersion) throws IOException {
-		String latestOclReleaseVersion = fetchLatestOclReleaseVersion(url, token);
-		if (lastReleaseVersion == null || !lastReleaseVersion.equals(latestOclReleaseVersion)) {
+	/**
+	 * This constructs the url to import concepts from the subscription url provided
+	 * @param url, the subscription url
+	 * @param version, desired version of the specified collection
+	 * @return GetMethod configured with the export url
+	 */
+	public GetMethod getExportUrl(String url, String version){
+		GetMethod exportUrlGet;
+
+		url = removeLastUrlForwardSlashIfExist(url);
+
+		if(url.contains(version)){
+			exportUrlGet = new GetMethod(url + "/export");
+		}else {
+			exportUrlGet = new GetMethod(url + "/" + version + "/export");
+		}
+		return exportUrlGet;
+	}
+
+	/**
+	 * This gets the desired collection release version
+	 * @param url, the subscription url
+	 * @param token, the subscription API token
+	 * @return the retrieved collection version
+	 * @throws IOException
+	 */
+	public String  getOclReleaseVersion(String url, String token) throws IOException {
+		String exportVersion;
+
+		url = removeLastUrlForwardSlashIfExist(url);
+
+		String subUrlAfterBaseUrl = url.substring(OCL_API_BASE_URL.length());
+

```

Comment:
If in all this you are simply trying to get the number of slashes, then wouldn't be both simpler and clearer if you used? StringUtils.countMatches()